cloudevents/pydantic/event.py

```diff
@@ -0,0 +1,240 @@
+    Supports both binary and structured mode CloudEvents
+    """
+
+    data: typing.Optional[typing.Any] = pydantic.Field(
+        title="Event Data",
+        description=(
+            "CloudEvents MAY include domain-specific information about the occurrence. "
+            "When present, this information will be encapsulated within data."
+            "It is encoded into a media format which is specified by the datacontenttype attribute "
+            "(e.g. application/json), and adheres to the dataschema "
+            "format when those respective attributes are present."
+        ),
+    )
+    source: event_attribute_types.NonEmptyURIReference = pydantic.Field(
+        title="Event Source",
+        description=(
+            "Identifies the context in which an event happened. "
+            "Often this will include information such as the type of the event source, "
+            "the organization publishing the event or the process that produced the event. "
+            "The exact syntax and semantics behind the data encoded in the URI is defined by the event producer.\n"
+            "\n"
+            "Producers MUST ensure that source + id is unique for each distinct event.\n"
+            "\n"
+            "An application MAY assign a unique source to each distinct producer, "
+            "which makes it easy to produce unique IDs since no other producer will have the same source. "
+            "The application MAY use UUIDs, URNs, DNS authorities or an application-specific "
+            "scheme to create unique source identifiers.\n"
+            "\n"
+            "A source MAY include more than one producer. In that case the producers "
+            "MUST collaborate to ensure that source + id is unique for each distinct event."
+        ),
+    )
+
+    id: pydantic.constr(min_length=1) = pydantic.Field(
+        default_factory=defaults.default_id_selection_algorithm,
+        title="Event ID",
+        description=(
+            "Identifies the event. Producers MUST ensure that source + id is unique for each distinct event. "
+            "If a duplicate event is re-sent (e.g. due to a network error) it MAY have the same id. "
+            "Consumers MAY assume that Events with identical source and id are duplicates. "
+            "MUST be unique within the scope of the producer"
+        ),
+    )
+    type: pydantic.constr(min_length=1) = pydantic.Field(
+        title="Event Type",
+        description=(
+            "This attribute contains a value describing the type of event related to the originating occurrence. "
+            "Often this attribute is used for routing, observability, policy enforcement, etc. "
+            "The format of this is producer defined and might include information such as the "
+            "version of the type"
+        ),
+    )
+    specversion: typing.Union[
+        typing.Literal["0.3"], typing.Literal["1.0"]
+    ] = pydantic.Field(
```

Comment:
IMO it's better to have a string enum with the spec versions. This way you can use them here as well as avoid `defaults` file.

Reply:
Agree about the enum. 

I dont really understand your point about the defaults file. Where should i put common default code between the http and pydantic module?